### PR TITLE
[POC] Testing qibolab 0.2

### PR DESCRIPTION
**Pull request by stavros11**
Updated the single shot classification routine to test with qibolab 0.2 (using qiboteam/qibolab#874).

I may try some other routines, but only few, as the pulse API for 0.2 is not finalized yet so it would be a waste of time to start updating routines now.

### CODE REVIEW

File: src/qibocal/protocols/characterization/coherence/t2_signal.py [hunk 1]
```diff
@@ -84,9 +86,9 @@ def _acquisition(
     )
 
     sweeper = Sweeper(
-        Parameter.start,
+        Parameter.duration,
         waits,
-        [RX90_pulses2[qubit] for qubit in targets],
+        [qd_delays[q] for q in targets] + [ro_delays[q] for q in targets],
```
[the remark this answers is not in the document]
Comment (alecandido) [reply src/qibocal/protocols/characterization/coherence/t2_signal.py:91]: The interface seems perfect to me.

For the duration discretization, I see it as a separate problem. Either we implicitly support padding and baking in the driver, or just raise an error when it is not possible to realize.

File: src/qibocal/protocols/characterization/coherence/t2_signal.py [hunk 2]
```diff
@@ -59,21 +59,23 @@ def _acquisition(
...
+        sequence.append(qd_delays[q])
+        sequence.append(qd_pulse)
+        sequence.append(
+            Delay(duration=2 * qd_pulse.duration, channel=qubit.readout.name)
         )
-        sequence.add(RX90_pulses1[qubit])
-        sequence.add(RX90_pulses2[qubit])
-        sequence.add(ro_pulses[qubit])
+        sequence.append(ro_delays[q])
+        sequence.append(ro_pulses[q])
```
Comment (alecandido) [inline src/qibocal/protocols/characterization/coherence/t2_signal.py:78]: You could make it even simpler:

```py
sequence.extend([qd_pulse, qd_delays[q], qd_pulse, Delay(duration=2 * qd_pulse.duration, channel=qubit.readout.name), ro_delays[q], ro_pulses[q]])
```

Both now and after the sequence layout update, the underlying structure is just a `list` and this is intended and part of the API. So, any `list` method is there to be used.

**Comment by stavros11** [comment]
@alecandido @hay-k I rebased this to latest qibocal main and updated some routines (resonator spectroscopy, Rabi, single shot classification and T1, T2 signal) for the new qibolab API, including the new channels, so that we can test and check how it looks from the user perspective. Here are some working examples with QM: https://github.com/qiboteam/qibolab/pull/933#issuecomment-2227109058

**Comment by alecandido** [comment]
Thanks @stavros11!

We don't need to have the full Qibocal working as soon as Qibolab 0.2 will be out, but being able to test a few common protocols (with different combination of features used) is definitely helpful.

File: src/qibocal/protocols/characterization/coherence/t2_signal.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (alecandido) [reply src/qibocal/protocols/characterization/coherence/t2_signal.py]: @stavros11 maybe we should close this comment, and open an issue in Qibolab for 0.2 for the drivers. What do you think?

**Pull request closed without merging**